pkg/event/dispatcher.go

```diff
@@ -159,7 +158,7 @@ func (ed *QueueEventDispatcher) flushEvents() {
 			ed.retryFlushCounter.Add(1)
 		}
 	}
-	ed.queueSize.Set(float64(ed.eventQueue.Size()))
+	ed.queueSize.Set(float64(queueSize))
```

Comment:
shouldn't it get the right size all the time when we use ed.eventQueue.Size() ? (just asking)

Reply:
`eventQueue.Size()` is a blocking call and I didn't feel it necessary to call it again here since it was just called from the for loop and we presumably exited with a size of zero. If the queue had changed since then it will be captured with the next call to `flushEvents()`.